byoeb-v1/byoeb/byoeb/apis/background_jobs.py

```
@@ -120,164 +115,56 @@ async def execute_job_function(job_function):
 def setup_scheduled_jobs():
     """Setup all scheduled jobs"""
     for job_config in JOB_CONFIGURATIONS:
-        if job_config["enabled"]:
+        if job_config.enabled:
             try:
                 scheduler.add_job(
                     execute_job_function,
-                    job_config["trigger"],
-                    args=[job_config["function"]],
-                    id=job_config["id"],
-                    name=job_config["name"],
+                    job_config.trigger,
+                    args=[job_config.function],
+                    id=job_config.id,
+                    name=job_config.name,
                     replace_existing=True,
                     misfire_grace_time=60
                 )
 
-                job_status[job_config["id"]] = {
-                    "status": "scheduled",
-                    "last_run": None,
-                    "error": None
-                }
-
-                _logger.info(f"Added job: {job_config['name']} with schedule: {job_config['trigger']}", extra={AppInsightsLogHandler.DETAILS: {
+                job_errors[job_config.id] = None
```

can we not remove this try catch part since it is blocking exception to propagate and give 5xx error?

---

Addressed by https://github.com/A4i-tech/byoeb/pull/79/commits/b2c01dc4b8bf36c2e2c8ae63980781234e4712d8 - the function is called during app lifespan but we can similarly benefit from fail fast here by causing deployment failure during staging for better exposure.